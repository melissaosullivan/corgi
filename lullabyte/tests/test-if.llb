/*
 * Tesing IF statements.
 */

void main(){
    
    int zero;
    int one;
    int two;
    boolean t;
    boolean f;

    zero = 0;
    one = 1;
    two = 2;
    t = true;
    f = false;

    /*****************************
     * Test Case:       if(true){}
     *
     * Result:          1
     */

    if (true) {
        print(1);
    }
    
    /*****************************
     * Test Case:       if(false){}
     *
     * Result:          [none]
     */

    if (false) {
        print(2);
    }

    /*****************************
     * Test Case:       if(true){} else{}
     *
     * Result:          3
     */

    if (true) {
        print(3);
    }
    else {
        print(4);
    }

    /*****************************
     * Test Case:       if(false){} else{}
     *
     * Result:          6
     */

    if (false) {
        print(5);
    }
    else {
        print(6);
    }

    /*****************************
     * Test Case:       if(evaluate equality){}
     *
     * Result:          true
     */

    if (t == !f) {
        print(true);
    }

    /*****************************
     * Test Case:       if(evaluate inequality){}
     *
     * Result:          false
     */

    if (f != !f) {
        print(false);
    }

    /*****************************
     * Test Case:       evaluating a binary operator to boolean
     *
     * Result:          42
     */

    if (two > one) {
        print(42);
    }

    /*****************************
     * Test Case:       setting a value before evaluation
     *
     * Result:          69
     */

    if (f = !f) {
        print(69);
    }

    /*****************************
     * Test Case:       if(integer)
     *
     * Result:          [none]
     */

    if (two) {
        print(69);
    }

    /*****************************
     * Test Case:       if(zero)
     *
     * Result:          [none]
     */

    if (zero) {
        print(69);
    }

}
