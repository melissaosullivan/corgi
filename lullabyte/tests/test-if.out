1
3
6
true
false
42
69
